app/models/coronavirus_restriction_search.rb

```diff
@@ -38,19 +35,54 @@ def invalid_postcode?
     %w[invalidPostcodeFormat fullPostcodeNoMapitValidation].include?(error_code)
   end
 
-  def no_restriction?
-    location_lookup.data.first&.england? && local_restriction.nil?
+  def devolved_nation?
+    ["Northern Ireland", "Scotland", "Wales"].include?(locations&.first&.country)
+  end
+
+  def england_result
+    return if error_code || locations.empty? || devolved_nation?
+
+    @england_result ||= begin
+      restriction_areas = locations.filter_map do |area|
+        CoronavirusRestrictionArea.find(area.gss)
```

Comment:
I'd forgotten what `CoronavirusRestrictionArea` is and had to go and look it up again. 
The confusion with the class would be reduced if the list was called `local_restrictions` rather than `restriction_areas` to remind us of what we we're getting back.